src/netlink.c

```diff
@@ -0,0 +1,362 @@
+	if(err!=IO_DONE){
+		lua_pushnil(L);
+		lua_pushliteral(L,"error sending message");
+		return 2;
+	}
+	lua_pushinteger(L,sent);
+	return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Receives data from a netlink socket
+\*-------------------------------------------------------------------------*/
+static int meth_receive(lua_State *L){
+	p_netlink nl= (p_netlink) auxiliar_checkclass(L,"netlink{connected}",1);
+	struct nlmsghdr *nlh;
+	size_t got;
+	p_timeout tm = &nl->tm;
+
+	nlh=(struct nlmsghdr *)malloc(NLMSG_SPACE(MAX_PAYLOAD));
+	memset(nlh,0,NLMSG_SPACE(MAX_PAYLOAD));
+	nlh->nlmsg_len=NLMSG_SPACE(MAX_PAYLOAD);
+
+	int err= socket_recv(&nl->fd,(char *)nlh,nlh->nlmsg_len,&got,tm);
+    	if (err != IO_DONE && err != IO_CLOSED) {
+        	lua_pushnil(L);
+       		lua_pushstring(L,socket_strerror(err));
+        	return 2;
+    	}
+    	lua_pushinteger(L,got);
+    	lua_pushlstring(L,NLMSG_DATA(nlh), got);
+    	return 2;
+}
+
+/*-------------------------------------------------------------------------*\
+* Receives data and sender from a netlink socket
+\*-------------------------------------------------------------------------*/
+static int meth_receivefrom(lua_State *L) {
+	p_netlink nl= (p_netlink) auxiliar_checkclass(L,"netlink{unconnected}",1);
+	struct sockaddr_nl dst;
+	struct nlmsghdr *nlh;
+	size_t got;
+	p_timeout tm = &nl->tm;
+
+	nlh=(struct nlmsghdr *)malloc(NLMSG_SPACE(MAX_PAYLOAD));
+	memset(nlh,0,NLMSG_SPACE(MAX_PAYLOAD));
+	nlh->nlmsg_len=NLMSG_SPACE(MAX_PAYLOAD);
+
+	socklen_t len=sizeof(dst);
+	int err= socket_recvfrom(&nl->fd,(char *)nlh,nlh->nlmsg_len,&got,(struct sockaddr *)&dst,&len,tm);
+    	if (err != IO_DONE && err != IO_CLOSED) {
+        	lua_pushnil(L);
+       		lua_pushstring(L,socket_strerror(err));
+        	return 2;
+    	}
+    	lua_pushinteger(L,got);
+    	lua_pushlstring(L,NLMSG_DATA(nlh), got);
+	lua_pushinteger(L,nlh->nlmsg_pid);
+    	return 3;
+}
+
+/*-------------------------------------------------------------------------*\
+* Select support methods
+\*-------------------------------------------------------------------------*/
+static int meth_getfd(lua_State *L) {
+    p_netlink nl = (p_netlink) auxiliar_checkgroup(L, "netlink{any}", 1);
+    lua_pushnumber(L, (int) nl->fd);
+    return 1;
+}
+
+/* this is very dangerous, but can be handy for those that are brave enough */
+static int meth_setfd(lua_State *L) {
+    p_netlink nl = (p_netlink) auxiliar_checkgroup(L, "netlink{any}", 1);
+    nl->fd = (t_socket) luaL_checknumber(L, 2);
+    return 0;
+}
+
+static int meth_getsockpid(lua_State *L){
+	p_netlink nl = auxiliar_checkgroup(L,"netlink{any}",1);
+	lua_pushinteger(L,nl->srcpid);
+	return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Binds an object to an address
+\*-------------------------------------------------------------------------*/
+static const char *netlink_trybind(p_netlink nl,int grp) {
```

Comment:
Are we sure `nl` is never null? If not, we should check it.

Reply:
meth_bind() (which is the function that calls trybind) already checks that through the auxiliar.h function auxiliar_checkclass()

Reply:
Also missing space